redfish-core/lib/log_services.hpp

```diff
@@ -841,18 +841,54 @@ inline void deleteDumpEntry(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
         "xyz.openbmc_project.Object.Delete", "Delete");
 }
 
-inline bool getDumpCompletionStatus(
-    std::vector<std::pair<std::string, std::variant<std::string>>>& values)
+inline int getDumpCompletionStatus(
+    std::vector<std::pair<std::string, std::variant<std::string>>>& values,
+    const std::shared_ptr<task::TaskData>& taskData, const std::string& objPath)
 {
     for (const std::pair<std::string, std::variant<std::string>>& statusProp :
          values)
     {
         if (statusProp.first == "Status")
         {
             const std::string& value = std::get<std::string>(statusProp.second);
-            return value !=
-                   "xyz.openbmc_project.Common.Progress.OperationStatus."
-                   "Completed";
+            if (value ==
+                "xyz.openbmc_project.Common.Progress.OperationStatus.Completed")
+            {
+                return 1;
+            }
+            if (value == "xyz.openbmc_project.Common.Progress."
+                         "OperationStatus.Failed")
+            {
+                return -1;
+            }
+            return 0;
+        }
+
+        if (statusProp.first == "DumpRequestStatus")
```

Comment:
This is hit for only resource dump right ?

Reply:
please add a comment saying that this will hit only for resource dump

Reply:
>  if (statusProp.first == "DumpRequestStatus")

Lets call this in a separate method once the Status=failed and for resource dump only

Reply:
> please add a comment saying that this will hit only for resource dump
done